model/task/task.go

```diff
@@ -497,19 +497,54 @@ func (t *Task) IsFinished() bool {
 // IsHostDispatchable returns true if the task should run on a host and can be
 // dispatched.
 func (t *Task) IsHostDispatchable() bool {
-	return (t.ExecutionPlatform == "" || t.ExecutionPlatform == ExecutionPlatformHost) && t.Status == evergreen.TaskUndispatched && t.Activated
+	return (t.ExecutionPlatform == "" || t.ExecutionPlatform == ExecutionPlatformHost) && len(t.ExecutionTasks) == 0 && t.Status == evergreen.TaskUndispatched && t.Activated
 }
 
 // IsContainerDispatchable returns true if the task should run in a container
 // and can be dispatched.
 func (t *Task) IsContainerDispatchable() bool {
-	return t.ExecutionPlatform == ExecutionPlatformContainer && t.Status == evergreen.TaskUndispatched && t.Activated && t.ContainerAllocated && t.Priority != evergreen.DisabledTaskPriority
+	if !t.ContainerAllocated {
```

Comment:
Can this logic be part of `isContainerScheduled`?

Reply:
I don't want to roll it into that one since `ShouldAllocateContainer` checks the opposite on L515.